k8s/airqo-deployments.yaml

```diff
@@ -0,0 +1,200 @@
+  type: NodePort
+
+##################################################################
+# Start data management microserive deployment and provisioning
+##################################################################
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: airqo-datamgt-service
+  namespace: staging
+  labels:
+    app: airqo-datamgt-service
+spec:
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: airqo-datamgt-service
+    spec:
+      containers:
+        - name: airqo-datamgt-service
+          image: us.gcr.io/airqo-250220/airqo-datamgt-service:latest
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 3000
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: airqo-datamgt-service-svc
+  namespace: staging
+spec:
+  selector:
+    app: airqo-datamgt-service
+  ports:
+    - protocol: TCP
+      port: 3000
+      targetPort: 3000
+      nodePort: 30002
+  type: NodePort
+
+###########################################################
+# Start ML microserive deployment and provisioning
+##########################################################
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: airqo-ml-service
+  namespace: staging
+  labels:
+    app: airqo-ml-service
+spec:
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: airqo-ml-service
+    spec:
+      containers:
+        - name: airqo-ml-service
+          image: us.gcr.io/airqo-250220/airqo-ml-service:latest
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 5000
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: airqo-ml-service-svc
+  namespace: staging
+spec:
+  selector:
+    app: airqo-ml-service
+  ports:
+    - protocol: TCP
+      port: 5000
+      targetPort: 5000
+      nodePort: 30003
+  type: NodePort
+###################################################################
+# Start analytics backend microserive deployment and provisioning
+###################################################################
+---
+apiVersion: apps/v1beta1
+kind: Deployment
+metadata:
+  name: airqo-analytics-backend
+  namespace: staging
+  labels:
+    app: airqo-analytics-backend
+spec:
+  replicas: 3
+  template:
+    metadata:
+      labels:
+        app: airqo-analytics-backend
+    spec:
+      containers:
+        - name: airqo-analytics-backend
+          image: us.gcr.io/airqo-250220/airqo-analytics-backend:latest
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 5000
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: airqo-analytics-backend-svc
+  namespace: staging
+spec:
+  selector:
+    app: airqo-analytics-backend
+  ports:
+    - protocol: TCP
+      port: 5000
+      targetPort: 5000
+      nodePort: 30004
+  type: NodePort
+
+####################################################################
+# Start analytics frontend microserive deployment and provisioning
```

Comment:
Thanks Daniel, good to see the PORTS configurations for the different services. Awesome stuff. 
I am just wondering about the inclusion of "analytics frontend" config details in this backend repo, could you shed some light on this?

Reply:
Ah ok, I was writing the scripts under the impression that all the K8S are supposed to be in a single file. I forgot to consider the repo differences :). Kindly advice! Should we create a "k8s" folder in the "AirQ0-frontend" repo?

Reply:
OoH, I get your point.

I am thinking that if we are to consider different build pipelines for both backend and frontend, we might need to created a k8s file for "AirQo-frontend". My imagination is that the tests for frontend are different from those of the frontend. Example: UI Selenium tests for the frontend are not required for the backend.....

Could you look into this further and share your new thoughts? Thanks again bro! :)

Reply:
Absolutely right, I think it will be appropriate to create a "k8s" folder in "AirQo-frontend" repo, which will contain the deployment scripts specific to frontend services. Nice catch :).